0011-interactive-transport-protocol/0011-interactive-transport-protocol.md

```diff
@@ -0,0 +1,104 @@
+
+Communicating payment requests from the receiver to the sender is *not* included in this protocol and is to be handled by higher level protocols.
+
+### Interfaces
+
+This protocol is used by higher-level protocols for the receiver to generate conditions and payment requests. The sender MAY modify or add to the request details but the payment request SHOULD serve as an Interledger packet without modification.
+
+### Operation
+
+The receiver creates an Interledger Packet from the details of the payment they want to request from the sender. The receiver uses a secret to create an HMAC of the packet. This HMAC is used as the condition fulfillment. The condition is the hash digest of the HMAC output.
+
+## Overview
+
+### Relation to Other Protocols
+
+     ------    
+    | SPSP |
+     ------ 
+       |
+     ------     -------
+    | ITP  |   |   ?   |
+     ------     -------
+       |           |
+     ------------------  
+    |       ILP        |
+     ------------------
+
+The [Simple Payment Setup Protocol (SPSP)](../0009-simple-payment-setup-protocol) uses the Interactive Transport Protocol (ITP). ITP is a host-to-host protocol intended for use on top of the Interledger Protocol.
+
+### Model of Operation
+
+We suppose that a sender wants to pay a receiver for some good or service and that the two are using a communication channel specified by a higher level protocol.
+
+We suppose that the receiver has a 32-byte secret value.
+
+1. The receiver creates an Interledger Packet using the details of the payment they want to request from the sender. The `condition` field is left undefined.
+2. The receiver creates the ITP Header (specified [below](#specification)) using a [UUID (version 4 is RECOMMENDED)](https://www.ietf.org/rfc/rfc4122.txt) and a token or string associated with the sender and/or good being purchased.
+3. The receiver attaches the ITP header to the Interledger Packet.
+4. The receiver encodes the complete Interledger Packet in the [Canonical OER](http://www.oss.com/asn1/resources/books-whitepapers-pubs/Overview%20of%20OER.pdf) format.
+5. The receiver creates an [HMAC](https://tools.ietf.org/html/rfc2104) of the encoded packet using the SHA-256 hash function and the receiver's secret as the HMAC key.
+6. The receiver uses the HMAC output as the _preimage_ for a [PREIMAGE-SHA-256 Crypto Condition](../0002-crypto-conditions) (the HMAC output is hashed again using SHA-256 to create the condition).
+7. The receiver uses this condition as the `condition` field in the Interledger Packet.
+8. The receiver communicates the complete Interledger Packet to the sender using the communication channel.
+9. The sender inspects the packet details.
+10. The sender requests a quote for the payment from a connector.
```

Comment:
You may also want to note that the sender should decide here whether they're ok with the cost that was quoted before proceeding.